### PR TITLE
Added Address endpoint

### PR DESCRIPTION
**Pull request by RogueElement**
Adds the following endpoints:

Info
Endpoint
| --- | --- |
| Summary of past 10 transactions involving address| `/address/X` |
| Raw info of past 10 transactions |  `/address/X/raw` |
| Summary with N transactions |  `/address/X/count/N` |
| Raw info with N past transactions | `address/X/count/N/raw` |

### CODE REVIEW

File: dcrsqlite/apisource.go [hunk 1]
```diff
@@ -468,3 +468,104 @@ func (db *wiredDB) GetMempoolSSTxDetails(N int) *apitypes.MempoolTicketDetails {
 	}
 	return &mpTicketDetails
 }
+
+//GetAddressTransactions returns an apitypes.Address Object
+//with at most the last 10 transactions the address was in
```
Comment (chappjc) [inline dcrsqlite/apisource.go]: comment reflects old 10 tx hardcode

File: dcrsqlite/apisource.go [hunk 2]
```diff
@@ -468,3 +468,104 @@ func (db *wiredDB) GetMempoolSSTxDetails(N int) *apitypes.MempoolTicketDetails {
...
+func (db *wiredDB) GetAddressTransactions(addr string, count int) *apitypes.Address {
+	address, err := dcrutil.DecodeAddress(addr)
+	if err != nil {
+		log.Errorf("Invalid address %s", addr)
```
Comment (chappjc) [inline dcrsqlite/apisource.go]: Also spit out the error:
`log.Errorf("Invalid address %s: %v", addr, err)`

File: dcrsqlite/apisource.go [hunk 3]
```diff
@@ -468,3 +468,104 @@ func (db *wiredDB) GetMempoolSSTxDetails(N int) *apitypes.MempoolTicketDetails {
...
+		return nil
+	}
+	txs, err := db.client.SearchRawTransactionsVerbose(address, 0, count, true, false, make([]string, 0))
```
Comment (chappjc) [inline dcrsqlite/apisource.go]: Can just pass a nil slice:

`txs, err := db.client.SearchRawTransactionsVerbose(address, 0, count, true, false, nil)`

File: dcrsqlite/apisource.go [hunk 4]
```diff
@@ -468,3 +468,104 @@ func (db *wiredDB) GetMempoolSSTxDetails(N int) *apitypes.MempoolTicketDetails {
...
+	if err != nil {
+		log.Errorf("GetAddressTransactions failed for address %s", addr)
```
Comment (chappjc) [inline dcrsqlite/apisource.go]: print `err` too

File: dcrsqlite/apisource.go [hunk 5]
```diff
@@ -468,3 +468,104 @@ func (db *wiredDB) GetMempoolSSTxDetails(N int) *apitypes.MempoolTicketDetails {
...
+		return nil
+	}
+	tx := make([]*apitypes.AddressTxShort, 0)
```
Comment (chappjc) [inline dcrsqlite/apisource.go]: If making a zero-length slice, it is more idiomatic to do:

```go
tx := []*apitypes.AddressTxShort{} // empty slice
// or
var tx []*apitypes.AddressTxShort // nil slice
```

These are technically different, but it rarely matters which you use.

However, if you are appending in a loop of known length, you should create a zero-length slice with the right capacity (for performance reasons):

```go
txarray := make([]*apitypes.AddressTxShort, 0, len(txs))
```

File: dcrsqlite/apisource.go [hunk 6]
```diff
@@ -468,3 +468,104 @@ func (db *wiredDB) GetMempoolSSTxDetails(N int) *apitypes.MempoolTicketDetails {
...
+	for i := range txs {
+		txarray := make([]apitypes.AddressTxValue, 0)
```
Comment (chappjc) [inline dcrsqlite/apisource.go]: same

File: dcrsqlite/apisource.go [hunk 7]
```diff
@@ -468,3 +468,104 @@ func (db *wiredDB) GetMempoolSSTxDetails(N int) *apitypes.MempoolTicketDetails {
...
+		Address:      addr,
+		Transactions: tx,
+	}
+
+}
+
+//GetAddressTransactions returns an array of apitypes.AddressTxRaw
+//objects representing the raw result of SearchRawTransactionsverbose
+func (db *wiredDB) GetAddressTransactionsRaw(addr string, count int) []*apitypes.AddressTxRaw {
+	address, err := dcrutil.DecodeAddress(addr)
```
Comment (chappjc) [inline dcrsqlite/apisource.go]: If you wipe out vendor/ and `glide install` from your branch, you are using older dependencies.  While a merge of the PR would handle this, it's best for you to be working on a branch that is really on top of master.  Also, when switching branches, it's best to always `rm -rf vendor` and `glide install`.

I saw a rebase master gave you some trouble, so we can go through this interactively if you want.  Resolving conflicts can be a pain without the right tools and expectations for what git is going to say at certain points.

File: dcrsqlite/apisource.go [hunk 8]
```diff
@@ -468,3 +468,104 @@ func (db *wiredDB) GetMempoolSSTxDetails(N int) *apitypes.MempoolTicketDetails {
...
+	if err != nil {
+		log.Errorf("Invalid address %s", addr)
```
Comment (chappjc) [inline dcrsqlite/apisource.go]: `err`

File: dcrsqlite/apisource.go [hunk 9]
```diff
@@ -468,3 +468,104 @@ func (db *wiredDB) GetMempoolSSTxDetails(N int) *apitypes.MempoolTicketDetails {
...
+		tx.Size = int32(len(txs[i].Hex) / 2)
+		tx.TxID = txs[i].Txid
+		tx.Version = txs[i].Version
+		tx.Locktime = txs[i].LockTime
+		tx.Vin = make([]dcrjson.VinPrevOut, len(txs[i].Vin))
+		copy(tx.Vin, txs[i].Vin)
+		tx.Confirmations = int64(txs[i].Confirmations)
+		tx.BlockHash = txs[i].BlockHash
+		tx.Blocktime = txs[i].Blocktime
+		tx.Time = txs[i].Blocktime
```
Comment (chappjc) [inline dcrsqlite/apisource.go]: You used `txs[i].Blocktime` twice on the right hand side.

File: dcrdataapi/apitypes.go [hunk 10]
```diff
@@ -104,6 +104,41 @@ type OutPoint struct {
...
+	Locktime      uint32               `json:"locktime"`
+	Vin           []dcrjson.VinPrevOut `json:"vin"`
+	Vout          []Vout               `json:"vout"`
+	Confirmations int64                `json:"confirmations"`
+	BlockHash     string               `json:"blockhash"`
+	Time          int64                `json:"time,omitempty"`
+	Blocktime     int64                `json:"blocktime,omitempty"`
+}
+
+//AddressTxShort is a subset of AddressTx with just the basic tx details pertaining the
```
Comment (chappjc) [inline dcrdataapi/apitypes.go]: subset of **AddressTxRaw**

File: dcrsqlite/apisource.go (hunk 7 above)
Comment (RogueElement) [inline dcrsqlite/apisource.go]: I'll try a rebase now to try and fix

**Pull request merged**